src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets

```diff
@@ -62,6 +62,10 @@ Copyright (c) .NET Foundation. All rights reserved.
     <WarningLevel Condition="'$(Language)' == 'C#' And
                              '$(AnalysisLevel)' == 'preview'">9999</WarningLevel>
 
+    <!-- Intermediate step to enable ILLink.Analyzers so ILLink, Blazor, Xamarin, AOT, etc. can enable the same flags -->
+    <AnalyzeSingleFile Condition="'$(PublishSingleFile)' == 'true'">true</AnalyzeSingleFile>
```

Comment:
Setting `AnalyzeSingleFile` but not `PublishSingleFile` won't produce any warnings, will it? I think we need to update the analyzers too: https://github.com/mono/linker/blob/9b71e7b1e0827aee8fb43baa1e599e35cdeeed03/src/ILLink.RoslynAnalyzer/SingleFileAnalyzer.cs#L55-L58

Also, what should happen if you set `<PublishSingleFile>true</PublishSingleFile>` and `<AnalyzeSingleFile>false</AnalyzeSingleFile>`? Do we want this to be settable independently of `PublishSingleFile`?

Reply:
You are right, there is a need to update the analyzers once this is merged so if AnalyzeSingleFile/AnalyzeTrimming gets set by another flag things still work.
The purpose of `AnalyzeSingleFile` is to have more conditions than just `PublishSingleFile`. All of these conditions will set to true `AnalyzeSingleFile` so there shouldn't be a case in which we end up with `PublishSingleFile` set to true and `AnalyzeSingleFile` set to other value different from `true` and if we ever end in that point because someone modified the property manually then the behavior is run publish single file without the analyzer running.